### ISSUE TITLE
Farming plot zone - QoL improvement features

### ISSUE THREAD

**Issue by Robik81** [CleverRaven/Cataclysm-DDA#24826]
## Farming plot zone ##
Aims to improve farming ease of use and to allow for further farming enhancements without making it less user friendly.

### Current state of farming ###
While farming is a great feature, it tends to be key press heavy and this problem seems to be increasing with time, rather than decreasing. In the near future, an **Improve crop farming PR** #24291 will be introduced and while it greatly enhances farming, some did express worries about excessive tediousness of such system.

I happen to agree that something has to be done to allow for such great improvement without killing all the fun in farming for the user.

### Quality of life features to help remedy the issue ###
- [x] designate a zone as a farming plot
- [x] action for tilling of nearby plots
- [x] implement support for the designating seed for the plot (wheat plot, strawberry plot, etc.) 
- [x] action for planting on nearby plots
- [ ] action for harvesting nearby plots
- [ ] watering plants
- [ ] fertilizing plants
- [ ] removing weeds
- [ ] ?

While I do have designating seed and planting marked as done, these would have to be updated, once #24291 is merged.

To give a better idea, how would such system work, I prepared short 1 minute video with my current implementation in action.

https://youtu.be/H8tBR3GdZLE

**Comment by nexusmrsep** [comment]
Looks sweet, great design idea.

**Comment by Vasyan2006** [comment]
I have planned to do something like thant in the future, but you already done most of the work.
Crop tending can be included relatively simple by separating watering and other actions into functions with plant tripoint as a parameter, and using them for every plant in the plot zone. These functions can be also used for NPC farms. I can do it, but I think it should wait until main farming PR will be merged.

**Comment by Robik81** [comment]
In my work on planting, I heavily refactored iexamine::dirtmound to pieces, so I could call the same code. If your watering etc. will be call from the outside ready, it will make the things easier for me. 

But yeah, no need to do it now. Let's wait until your PR is merged.

**Comment by Nightquaker** [comment]
I'm not a particular fan of such changes, since they basically mean "removing actual work from a thing that is supposed to simulate work". Especially in such a realistic (relatively speaking) game as CDDA. But I am excited for more depth enhancements of the farming aspect as a whole. So I'm happy with the changes overall.

**Comment by Nightquaker** [comment]
Ideally, I really like when such features are optional.
People who don't have much time to spare and want to avoid tedium use the feature like that to get around faster.
And people who do have time to spare and want to simulate the real-life tedium of manual farming, would just not have to use the feature that automates it all.

**Comment by prutschman** [comment]
Pull request #27278 contains a candidate implementation of zone activity commands for harvesting and fertilization.

**Comment by davidpwbrown** [comment]
Closing as this is now addressed by zone farming activities, which now includes harvesting, and fertilizing.
If watering and weeding is required in future, the framework for the zone activities should allow for that fairly easily.
